toolchain/check/testdata/where_expr/fail_not_facet.carbon

```diff
@@ -12,11 +12,40 @@
 
 library "[[@TEST_NAME]]";
 
-// CHECK:STDERR: fail_left_where_not_facet.carbon:[[@LINE+3]]:10: error: left argument of `where` operator must be a facet type [WhereOnNonFacetType]
+// CHECK:STDERR: fail_left_where_not_facet.carbon:[[@LINE+4]]:10: error: left argument of `where` operator must be a facet type [WhereOnNonFacetType]
 // CHECK:STDERR: fn F(T:! i32 where .Self == bool);
 // CHECK:STDERR:          ^~~
+// CHECK:STDERR:
 fn F(T:! i32 where .Self == bool);
 
+// --- fail_left_where_unknown.carbon
+
+library "[[@TEST_NAME]]";
+
+// CHECK:STDERR: fail_left_where_unknown.carbon:[[@LINE+8]]:10: error: name `NOT_DECLARED` not found [NameNotFound]
+// CHECK:STDERR: fn G(U:! NOT_DECLARED where .Self == bool);
+// CHECK:STDERR:          ^~~~~~~~~~~~
+// CHECK:STDERR:
+// CHECK:STDERR: fail_left_where_unknown.carbon:[[@LINE+4]]:10: error: left argument of `where` operator must be a facet type [WhereOnNonFacetType]
+// CHECK:STDERR: fn G(U:! NOT_DECLARED where .Self == bool);
+// CHECK:STDERR:          ^~~~~~~~~~~~
+// CHECK:STDERR:
+fn G(U:! NOT_DECLARED where .Self == bool);
+
+
+// --- fail_var.carbon
+
+library "[[@TEST_NAME]]";
+
+// CHECK:STDERR: fail_var.carbon:[[@LINE+7]]:8: error: name `e` not found [NameNotFound]
+// CHECK:STDERR: var v: e where .x = 3;
+// CHECK:STDERR:        ^
+// CHECK:STDERR:
+// CHECK:STDERR: fail_var.carbon:[[@LINE+3]]:8: error: left argument of `where` operator must be a facet type [WhereOnNonFacetType]
+// CHECK:STDERR: var v: e where .x = 3;
+// CHECK:STDERR:        ^
+var v: e where .x = 3;
```

Comment:
Could you also add a test for a valid top-level `where`?

Reply:
Added to `constraints.carbon`